utils/util_common.py

```diff
@@ -60,3 +61,34 @@ def kwargs_from_props(props):
             # property group within this property
             result.update(kwargs_from_props(prop))
     return result
+
+
+def resitriced_size(parent_dimensions, offset, size_min, size):
+    """ Get size restricted by various factors
+    """
+    limit_x = min(parent_dimensions[0] + 2*offset[0], parent_dimensions[0] - 2*offset[0])
+    limit_y = min(parent_dimensions[1] + 2*offset[1], parent_dimensions[1] - 2*offset[1])
+    x = max(min(limit_x, size[0]), size_min[0])
```

Comment:
Should use already existing `clamp` function in `util_common.py` for the expression
```python 
max(min(value, minimum, maximum)
```
Improves code clarity.